spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/RibbonCommand.java

```diff
@@ -80,7 +80,7 @@ public RibbonCommand(String commandKey, RestClient restClient, Verb verb, String
 		super(getSetter(commandKey));
 		this.restClient = restClient;
 		this.verb = verb;
-		this.uri = (StringUtils.hasText(uri))? UriComponentsBuilder.fromUriString(uri).build().toUri() : new URI(uri);
```

Comment:
The only thing I did here was remove the escaping logic to allow for it to occur outside the ribbon command. The logic in the original would still throw NPE if the RibbonCommand was constructed with a null URI string. 

Does it make more sense to do a null check or make a non-null uri a design constraint with an assertion?